src/uu/cp/src/cp.rs

```diff
@@ -931,23 +931,7 @@ impl Options {
         };
 
         // Parse attributes to preserve
-        let mut attributes =
-            if let Some(attribute_strs) = matches.get_many::<String>(options::PRESERVE) {
-                if attribute_strs.len() == 0 {
-                    Attributes::DEFAULT
-                } else {
-                    Attributes::parse_iter(attribute_strs)?
-                }
-            } else if matches.get_flag(options::ARCHIVE) {
-                // --archive is used. Same as --preserve=all
-                Attributes::ALL
-            } else if matches.get_flag(options::NO_DEREFERENCE_PRESERVE_LINKS) {
-                Attributes::LINKS
-            } else if matches.get_flag(options::PRESERVE_DEFAULT_ATTRIBUTES) {
-                Attributes::DEFAULT
-            } else {
-                Attributes::NONE
-            };
+        let mut attributes = Self::determine_attributes(matches)?;
 
         // handling no-preserve options and adjusting the attributes
         if let Some(attribute_strs) = matches.get_many::<String>(options::NO_PRESERVE) {
```

Comment:
This `if`-block and the next one can be removed, they have been moved to `handle_no_preserve_option` and `determine_attributes`, respectively.